src/components/PaymentEditDialog.js

```diff
@@ -0,0 +1,201 @@
+import {
+    MuiPickersUtilsProvider,
+    KeyboardDatePicker
+} from '@material-ui/pickers'
+import CurrencyTextField from '@unicef/material-ui-currency-textfield'
+import moment from 'moment'
+import MomentUtils from '@date-io/moment'
+
+import LoadingProgress from './LoadingProgress'
+import { EDIT_PAYMENT } from '../operations/mutations/PaymentMutations'
+import {
+    selectCurrencyInformation
+} from '../scripts/selectors'
+
+const PaymentEditDialog = (props) => {
+
+    const {
+        payment,
+        onClose,
+        open
+    } = props
+
+    const currencyInformation = selectCurrencyInformation({
+        currency: payment.client.currency
+    })
+    const history = useHistory()
+
+    const [editPayment, {
+        dataPayment,
+        loadingPayment,
+        errorPayment
+    }] = useMutation(EDIT_PAYMENT)
+
+    const [dateIncurred, setDateIncurred] = useState(null)
+    const [datePaid, setDatePaid] = useState(null)
+    const [disableEdit, setDisableEdit] = useState(true)
+    const [displayError, setDisplayError] = useState(false)
+    const [editPaymentError, setEditPaymentError] = useState('')
+    const [invalidPaymentAmountInput, setInvalidPaymentAmountInput] = useState(false)
+    const [paymentAmount, setPaymentAmount] = useState(null)
+
+    const handleAlertClose = (event, reason) => {
+        if (reason === 'clickaway') {
+            return
+        }
+        setDisplayError(false)
+    }
+    const handleEditPayment = async () => {
+        const variables = {
+            id: Number(payment.id),
+            amount: paymentAmount,
+            date_incurred: dateIncurred,
+            date_paid: datePaid
+        }
+        const editedPayment = await editPayment({
+            variables: variables
+        })
+        if (loadingPayment) return <LoadingProgress/>
+        if (editedPayment.errors) {
+            setEditPaymentError(`${Object.keys(editedPayment.errors[0].extensions.exception.fields)[0]}`)
+            setDisplayError(true)
+        } else {
+            onClose()
+
+        }
+    }
+    const handleDateIncurredChange = (date) => {
+        setDateIncurred(moment(date['_d']).format('YYYY-MM-DD'))
+    }
+    const handleDatePaidChange = (date) => {
+        setDatePaid(moment(date['_d']).format('YYYY-MM-DD'))
+    }
+    const handlePaymentAmountChange = (input) => {
+        setInvalidPaymentAmountInput(false)
+        const amount = typeof input == 'string' ? Number(input.replace(/\D/g, '')) : (input / 100)
+        setPaymentAmount(amount)
+    }
+
+    useEffect(() => {
+        if (!dateIncurred || !paymentAmount) {
+            setDisableEdit(true)
+        } else if (
+            dateIncurred == moment(payment.date_incurred, 'x').format('YYYY-MM-DD') &&
+            datePaid == moment(payment.date_paid, 'x').format('YYYY-MM-DD')
+            && paymentAmount == payment.amount / 100) {
+            setDisableEdit(true)
+        } else {
+            setDisableEdit(false)
+        }
```

Comment:
Got confused at this logic block because of the improper indent. Should be:

```
if (!dateIncurred || !paymentAmount) {
    setDisableEdit(true)
} else if (
    dateIncurred == moment(payment.date_incurred, 'x').format('YYYY-MM-DD') &&
    datePaid == moment(payment.date_paid, 'x').format('YYYY-MM-DD')
    && paymentAmount == payment.amount / 100
) {
    setDisableEdit(true)
} else {
    setDisableEdit(false)
}
```

Otherwise the function call looks like a 4th condition of the `else if`